### DISCUSSION TITLE
Vercel Build API output dir

### DISCUSSION THREAD

**Discussion by melMass** [vercel/community#904] [Help] [answered]
Trying a new one as for some reason the discussion was locked: https://github.com/orgs/vercel/discussions/894

When using Vercel Build API, what is the correct « output dir »?

Thanks

**Comment by amyegan** [comment, marked as answer]
Hi @melMass! When using the [Build Output API](https://vercel.com/docs/build-output-api/v3), files should be output to the `.vercel/output` directory on build. 

Going back to your previous post, the framework presets generally handle output directory without the need to override unless you changed the default outputs for the framework build. 

Depending on what [monorepo](https://vercel.com/docs/concepts/monorepos) tools you are using, there may be some additional configuration necessary. I can share more specific resources for monorepo deployments if you need them. Just let me know what tools you are using 🙂

**Reply by melMass** [reply]
Thanks! I'm using [NX](https://nx.dev/).
I did found that and tried to set the output directory directly to `apps/my-app/.vercel/output` but that gave me this (I enabled directory listing to debug the issue)
![image](https://user-images.githubusercontent.com/7041726/192818437-115bd3ed-c209-4531-83ee-344e82fa9084.png)

For now this works, but maybe there is a better way:

> 1. set the root in vercel to `apps/my-app`
> 2. set the build command to (adapt to your tooling): 
    `cd ../.. && npx nx build my-app`
untick the output dir to use the default one 
    (it still states dist but it works at least!)

**Reply by amyegan** [reply]
We have a document that walks through [deploying Nx projects to Vercel](https://vercel.com/docs/concepts/monorepos/nx#deploy-to-vercel) should help. 

There is a step in those instructions about [setting the output directory](https://vercel.com/docs/concepts/monorepos/nx#set-the-output-directory). The screenshot example assumes Next.js, but you would use your app's project.json [outputs](https://nx.dev/reference/project-configuration#outputs) setting for that. And then you would set the root directory to either `./` or whatever other directory is the root of your monorepo where `nx.json` lives.
